front/temporal/labs/connections/providers/freshservice/client.ts

```diff
@@ -0,0 +1,370 @@
+import { isLeft } from "fp-ts/lib/Either";
+import * as reporter from "io-ts-reporters";
+import { freshServiceLimiter } from "@app/temporal/labs/connections/providers/freshservice/utils";
+import {
+  ConversationResponseCodec,
+  SlaPolicyResponseCodec,
+  TaskResponseCodec,
+  TicketCodec,
+  TicketResponseCodec,
+  AssetCodec,
+  ProblemCodec,
+  ChangeCodec,
+} from "@app/temporal/labs/connections/providers/freshservice/types";
+
+export class FreshServiceError extends Error {
+  readonly status?: string;
+  readonly endpoint: string;
+  readonly body?: {
+    code?: string;
+    message?: string;
+  };
+
+  constructor(
+    message: string,
+    {
+      endpoint,
+      status,
+      body,
+    }: {
+      endpoint: string;
+      status?: string;
+      body?: {
+        code?: string;
+        message?: string;
+      };
+    }
+  ) {
+    super(message);
+    this.endpoint = endpoint;
+    this.status = status;
+    this.body = body;
+  }
+
+  static fromValidationError({
+    endpoint,
+    pathErrors,
+  }: {
+    endpoint: string;
+    pathErrors: string[];
+  }) {
+    return new this("Response validation failed", {
+      endpoint,
+      body: {
+        code: "validation_error",
+        message: pathErrors.join(", "),
+      },
+    });
+  }
+}
+
+export class FreshServiceClient {
+  public readonly baseURL: string;
+
+  constructor(
+    private readonly apiKey: string,
+    private readonly domain: string
+  ) {
+    this.baseURL = `https://${domain}.freshservice.com`;
+    logger.info({ domain: this.baseURL }, "FreshService client initialized");
+  }
+
+  private async makeRequest<T>(
+    endpoint: string,
+    codec: t.Type<T>,
+    options: {
+      method?: string;
+      params?: Record<string, string | number>;
+      body?: any;
+    } = {}
+  ): Promise<T> {
+    const { method = "GET", params = {}, body } = options;
+
+    // Build URL with query parameters
+    let url = `${this.baseURL}${endpoint}`;
+    if (Object.keys(params).length > 0) {
+      const queryParams = new URLSearchParams();
+      Object.entries(params).forEach(([key, value]) => {
+        queryParams.append(key, String(value));
+      });
+      url += `?${queryParams.toString()}`;
+    }
+
+    // Freshservice counts an additional call for each param in the include query param
+    // so we need to calculate the weight of the request based on the number of fields in the include query param
+    const include = params.include as string;
+    const includeFields = include ? include.split(",") : [];
+    const weight = includeFields.length + 1;
+
+    const parsedUrl = new URL(url);
+
+    const requestOptions = {
+      hostname: parsedUrl.hostname,
+      port: 443,
+      path: `${parsedUrl.pathname}${parsedUrl.search}`,
+      method,
+      headers: {
+        "Content-Type": "application/json",
+        Authorization:
+          "Basic " + Buffer.from(`${this.apiKey}:X`).toString("base64"),
+      } as Record<string, string>,
```

Comment:
`nit`
Rather than type assertion you might want to do:
```suggestion
    const requestOptions: Record<string, string> = {
      hostname: parsedUrl.hostname,
      port: 443,
      path: `${parsedUrl.pathname}${parsedUrl.search}`,
      method,
      headers: {
        "Content-Type": "application/json",
        Authorization:
          "Basic " + Buffer.from(`${this.apiKey}:X`).toString("base64"),
      },
```